src/core/lwt.ml

```diff
@@ -3133,6 +3133,11 @@ struct
   let (<&>) p p' = join [p; p']
   let (<?>) p p' = choose [p; p']
 
+end
+include Infix
+
+module Let_syntax =
```

Comment:
I made this change as when i try to use the ppx in the form `let%bind.Lwt`, it looks for a module `Lwt.Let_syntax.Let_syntax`. Allowing for `let%bind.<module>` helps when mixing/matching different monads.

Reply:
Could you briefly explain/link us to why ppx_let expects duplicate nesting like this? I understand it's likely intentional, but it seems like some sort of bug.

Reply:
> Could you briefly explain/link us to why ppx_let expects duplicate nesting like this? I understand it's likely intentional, but it seems like some sort of bug.

I tried to match the monad interface defined in the base library (which is what i believe `ppx_let` targets).

https://github.com/janestreet/base/blob/7fb12eef1e8eb23856f3337e47dbe7522c1955dd/src/monad_intf.ml#L44-L59

I'm not too familiar with the ppx implementation, but when I tried to use the `ppx_let` with module resolution via `let%bind.Lwt`, it complained about not finding a module `Let_syntax.Let_syntax` when i didn't have this nesting. I do agree it looks a little strange. I think i'll be fine with not having this nesting as the only thing that'll not work is mixing monads and using module resolution to tell the ppx about which one to use:

```ocaml
let%bind.Lwt x = x in
let%bind.Lwt_result y = y
...
```